pip/vcs/subversion.py

```diff
@@ -263,6 +265,24 @@ def check_version(self, dest, rev_options):
         """Always assume the versions don't match"""
         return False
 
+    @staticmethod
+    def remove_auth_from_url(url):
+        # Return a copy of url with 'username:password@' removed.
+        # username/pass params are passed to subversion through flags
+        # and are not recognized in the url.
+
+        # parsed url
+        purl = urllib_parse.urlsplit(url)
+        stripped_netloc = \
+            purl.netloc.split('@')[-1] if '@' in purl.netloc else purl.netloc
+
+        # stripped url
+        url_pieces = (
+            purl.scheme, stripped_netloc, purl.path, purl.query, purl.fragment
+        )
+        surl = urllib_parse.urlunsplit(url_pieces)
```

Comment:
It helps anyone debugging in the future that needs to step through the code to see the return value while still in the same stack frame.